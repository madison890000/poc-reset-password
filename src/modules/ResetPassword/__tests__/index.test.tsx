import React from 'react';
import ResetPasswordModule from '../index';
import renderer from 'react-test-renderer';
import {fireEvent, render, screen, waitFor} from '@testing-library/react';

describe('renders ResetPasswordModule initial', () => {
    test('snapshot for ResetPasswordModule initial', () => {
        const resetPasswordModule = renderer.create(<ResetPasswordModule/>).toJSON();
        expect(resetPasswordModule).toMatchSnapshot();
    });
    test('ResetPasswordModule should have a input', () => {
        render(<ResetPasswordModule/>);
        const passwordInput = screen.getByTestId('reset-password');
        expect(passwordInput).toBeVisible();
    });
    test('ResetPasswordModule should have a reset button', () => {
        render(<ResetPasswordModule/>);
        const passwordResetButton = screen.getByTestId('reset-btn');
        expect(passwordResetButton).toBeVisible();
    });

    test('ResetPasswordModule should have a reset button and disabled', () => {
        render(<ResetPasswordModule/>);
        const passwordResetButton = screen.getByTestId('reset-btn');
        expect(passwordResetButton.getAttribute('disabled')).toBe('');
    });
});


describe('ResetPasswordModule word good', () => {

    test('input password correctly, ResetPasswordModule should have a reset button and enabled', async () => {
        render(<ResetPasswordModule/>);
        const passwordResetButton = screen.getByTestId('reset-btn');
        expect(passwordResetButton.getAttribute('disabled')).toBe('');
        const passwordInput = screen.getByTestId('reset-password');
        fireEvent.change(passwordInput, {target: {value: 'abc123__abd'}});
        await waitFor(() => {
            const passwordResetButton = screen.getByTestId('reset-btn');
            // no disable attribute
            expect(passwordResetButton.getAttribute('disabled')).toBe(null);
        })
    });
    test('input password wrong, ResetPasswordModule should have a reset button and enabled', async () => {
        render(<ResetPasswordModule/>);
        const passwordResetButton = screen.getByTestId('reset-btn');
        expect(passwordResetButton.getAttribute('disabled')).toBe('');
        const passwordInput = screen.getByTestId('reset-password');
        fireEvent.change(passwordInput, {target: {value: 'abc123'}});
        await waitFor(() => {
            const passwordResetButton = screen.getByTestId('reset-btn');
            // has disable attribute
            expect(passwordResetButton.getAttribute('disabled')).toBe('');
        })
    });
});