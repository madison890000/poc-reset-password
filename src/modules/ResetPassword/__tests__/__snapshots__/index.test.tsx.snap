// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResetPasswordModule initial snapshot for ResetPasswordModule initial 1`] = `
<div>
  <div>
    Reset Password
  </div>
  <div>
    <input
      data-testid="reset-password"
      onChange={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "color": "red",
      }
    }
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="reset-btn"
    disabled={true}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    Reset Your Password
  </button>
</div>
`;
